mesonbuild/modules/python3.py

```diff
@@ -38,35 +40,37 @@ class Python3Module(ExtensionModule):
 
     INFO = ModuleInfo('python3', '0.38.0', deprecated='0.48.0')
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, interpreter: Interpreter):
+        super().__init__(interpreter)
         self.methods.update({
             'extension_module': self.extension_module,
             'find_python': self.find_python,
             'language_version': self.language_version,
             'sysconfig_path': self.sysconfig_path,
         })
 
-    @permittedKwargs(known_shmod_kwargs - {'name_prefix', 'name_suffix'})
```

Comment:
What???

Also just want to reiterate that this functional change cannot be validly described by the commit message "Port to the new module API and add it to mypy."

Reply:
Added this back and folded most of the commit into other commits which more match what the changes actually are.